src/cdk/explorer/nodes/appNode.ts

```diff
@@ -0,0 +1,82 @@
+/*!
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as nls from 'vscode-nls'
+const localize = nls.loadMessageBundle()
+
+import * as path from 'path'
+import * as vscode from 'vscode'
+import { AWSTreeNodeBase } from '../../../shared/treeview/nodes/awsTreeNodeBase'
+import { PlaceholderNode } from '../../../shared/treeview/nodes/placeholderNode'
+import { cdk } from '../../globals'
+import { CdkAppLocation, getApp } from '../cdkProject'
+import { includeConstructInTree } from '../tree/treeInspector'
+import { ConstructNode } from './constructNode'
+
+/**
+ * Represents a CDK App
+ * Existence of apps is determined by the presence of `cdk.json` in a workspace folder
+ */
+export class AppNode extends AWSTreeNodeBase {
+    get tooltip(): string {
+        return this.app.cdkJsonPath
+    }
+
+    get id(): string {
+        return this.app.treePath
+    }
+
+    public constructor(public readonly app: CdkAppLocation) {
+        super(app.cdkJsonPath, vscode.TreeItemCollapsibleState.Collapsed)
+        this.contextValue = 'awsCdkAppNode'
+        this.label = path.relative(app.workspaceFolder.uri.fsPath, app.cdkJsonPath)
+
+        this.iconPath = {
+            dark: vscode.Uri.file(cdk.iconPaths.dark.cdk),
+            light: vscode.Uri.file(cdk.iconPaths.light.cdk)
+        }
+    }
+
+    public async getChildren(): Promise<(ConstructNode | PlaceholderNode)[]> {
+        const constructs = []
+        try {
+            const successfulApp = await getApp(this.app)
+
+            const constructsInTree = successfulApp.metadata.tree.children
+            if (constructsInTree) {
+                for (const construct of Object.keys(constructsInTree)) {
+                    const entity = constructsInTree[construct]
+                    if (includeConstructInTree(entity)) {
+                        constructs.push(
+                            new ConstructNode(
+                                entity.id,
+                                entity.children
+                                    ? vscode.TreeItemCollapsibleState.Collapsed
+                                    : vscode.TreeItemCollapsibleState.None,
+                                entity
+                            )
+                        )
+                    }
+                }
+            }
+
+            // indicate that App exists, but it is empty
+            if (constructs.length === 0) {
+                return [
+                    new PlaceholderNode(this, localize('Aws.cdk.explorerNode.app.noStacks', '[no stacks in this app]'))
```

Comment:
Sanity checking terminology here... should it say "no stacks" or "no constructs" ?
Also: Casing... consider `[No Stacks in this CDK App]`

Reply:
both are true. The CLI currently displays "no stacks" so I aligned with that.

Reply:
Ok